backend-lib/src/main/rust/lib.rs

```diff
@@ -80,16 +80,16 @@ fn print_debug_state() {
 }
 
 fn wallet_db<P: Parameters>(
-    env: &JNIEnv<'_>,
+    env: &mut JNIEnv,
```

Comment:
I'm surprised this wasn't &mut already, given that [a `JNIEnv` can only be used by a single thread](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html#attaching_to_the_vm).